### ISSUE TITLE
Bug: sam local start-api with debug port runs into TypeError

### ISSUE THREAD

**Issue by darhsu** [aws/aws-sam-cli#6053]
<!-- Make sure we don't have an existing Issue that reports the bug you are seeing (both open and closed). 
If you do find an existing Issue, re-open or add a comment to that Issue instead of creating a new one. -->

### Description:
<!-- Briefly describe the bug you are facing.-->
When running `sam local start-api` with `debug_port = 5858` in the `samconfig.toml` file, I get a `TypeError: 'Integer' object is not iterable`.

However, there's no error when running `sam local start-api --debug-port 5858` when adding the --debug-port flag directly in the command. 


### Steps to reproduce:
<!-- Provide detailed steps to replicate the bug, including steps from third party tools (CDK, etc.) --> 
1. `sam init` # create hello world example with Python and zip
2. Add `debug_port = 5858` in `samconfig.toml`:
``` toml
[default.local_start_api.parameters]
warm_containers = "EAGER"
debug_port = 5858
```
3. `sam build`
4. `sam local start-api`



### Observed result:
<!-- Please provide command output with `--debug` flag set. -->
``` bash
/workspaces/sam-app $ sam local start-api --debug
2023-10-09 05:11:18,765 | Config file location: /workspaces/sam-app/samconfig.toml                                                                                                                                                
2023-10-09 05:11:18,769 | Loading configuration values from [default.['local', 'start-api'].parameters] (env.command_name.section) in config file at '/workspaces/sam-app/samconfig.toml'...                                      
2023-10-09 05:11:18,770 | Configuration values successfully loaded.                                                                                                                                                               
2023-10-09 05:11:18,771 | Configuration values are: {'stack_name': 'sam-app', 'warm_containers': 'EAGER', 'debug_port': 5858}                                                                                                     
2023-10-09 05:11:18,775 | Using SAM Template at /workspaces/sam-app/.aws-sam/build/template.yaml                                                                                                                                  
Traceback (most recent call last):
  File "samcli/__main__.py", line 12, in <module>
  File "click/core.py", line 1157, in __call__
  File "click/core.py", line 1078, in main
  File "click/core.py", line 1688, in invoke
  File "click/core.py", line 1686, in invoke
  File "click/core.py", line 943, in make_context
  File "click/core.py", line 1408, in parse_args
  File "click/core.py", line 2396, in handle_parse_result
  File "click/core.py", line 2950, in consume_value
TypeError: 'Integer' object is not iterable
[14390] Failed to execute script '__main__' due to unhandled exception!
```



### Expected result:
<!-- Describe what you expected. -->
Running `sam local start-api --debug-port 5858` with the debug port directly in the command:
``` bash
/workspaces/sam-app $ sam local start-api --debug-port 5858
Initializing the lambda functions containers.                                                                                                                                                                                     
Local image was not found.                                                                                                                                                                                                        
Removing rapid images for repo public.ecr.aws/sam/emulation-python3.10                                                                                                                                                            
Building image....................
Using local image: public.ecr.aws/lambda/python:3.10-rapid-x86_64.                                                                                                                                                                
                                                                                                                                                                                                                                  
Mounting /workspaces/sam-app/.aws-sam/build/HelloWorldFunction as /var/task:ro,delegated, inside runtime container                                                                                                                
Containers Initialization is done.                                                                                                                                                                                                
Mounting HelloWorldFunction at http://127.0.0.1:3000/hello [GET]                                                                                                                                                                  
You can now browse to the above endpoints to invoke your functions. You do not need to restart/reload SAM CLI while working on your functions, changes will be reflected instantly/automatically. If you used sam build before    
running local commands, you will need to re-run sam build for the changes to be picked up. You only need to restart SAM CLI if you update your AWS SAM template                                                                   
2023-10-09 05:09:17 WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:3000
``` 


### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

<!-- Either provide the following info (for AWS SAM CLI v1.68.0 or before) or paste the output of `sam --info` (AWS SAM CLI v1.69.0 or after). -->

1. OS: Ubuntu
2. `sam --version`: 1.98.0
3. AWS region: us-east-1


### Paste the output of `sam --info` here
``` json
{
  "version": "1.98.0",
  "system": {
    "python": "3.11.3",
    "os": "Linux-5.15.0-1041-azure-x86_64-with-glibc2.31"
  },
  "additional_dependencies": {
    "docker_engine": "23.0.6+azure-2",
    "aws_cdk": "Not available",
    "terraform": "Not available"
  },
  "available_beta_feature_env_vars": [
    "SAM_CLI_BETA_FEATURES",
    "SAM_CLI_BETA_BUILD_PERFORMANCE",
    "SAM_CLI_BETA_TERRAFORM_SUPPORT",
    "SAM_CLI_BETA_RUST_CARGO_LAMBDA"
  ]
}
```

**Comment by sriram-mv** [comment]
Thanks for the bug report! This is certainly error messaging that needs to be fixed.

There seems to be a miss in functionality that needs to be surfaced. debug port is actually not a single port, one can specify a list of ports for debugging.

```
[default.local_start_api.parameters]
warm_containers = "EAGER"
debug_port = ["5858", "5851"]
```

should work from the configuration file.

This will open multiple ports from host to container for communication.

**Comment by darhsu** [comment]
@sriram-mv Thanks for your help! I can confirm that setting `debug_port` to a list of strings in the configuration file works!

**Comment by mildaniel** [comment]
This issue has now been resolved in version 1.100.0.
